main.go

```diff
@@ -0,0 +1,205 @@
+		return &resource.Domain{}, "", routeUrl, fmt.Errorf("please provide the url including the scheme")
+	}
+
+	domains, err := retrieveDomains(cfc, routeUrl.Hostname())
+	if err != nil {
+		return &resource.Domain{}, routeUrl.Hostname(), routeUrl, err
+	}
+
+	if len(domains) == 0 {
+		urlParts := strings.SplitN(routeUrl.Hostname(), ".", 2)
+		if len(urlParts) < 2 {
+			return &resource.Domain{}, "", routeUrl, fmt.Errorf("'%s' is not a domain", routeUrl.Hostname())
+		}
+		hostName := urlParts[0] //Subdomain is not empty
+		domainName := urlParts[1]
+		domains, err = retrieveDomains(cfc, domainName)
+		if len(domains) == 0 {
+			return &resource.Domain{}, hostName, routeUrl, fmt.Errorf("error retrieving apps: route not found, domain '%s' is unknown", domainName)
+		}
+		return domains[0], hostName, routeUrl, nil
+	}
+	return domains[0], routeUrl.Hostname(), routeUrl, nil
+}
+
+func findRoute(cfc *client.Client, query string) (*resource.Route, error) {
+	domain, hostName, routeUrl, err := findDomain(cfc, query)
+
+	opts := client.NewRouteListOptions()
+	opts.Hosts.Values = append(opts.Hosts.Values, hostName)
+	opts.DomainGUIDs.Values = append(opts.DomainGUIDs.Values, domain.GUID)
+	opts.Paths.Values = append(opts.Paths.Values, routeUrl.Path)
+
+	routes, err := cfc.Routes.ListAll(context.Background(), opts)
+	if err != nil {
+		return &resource.Route{}, err
+	}
+
+	if len(routes) == 0 { // Wildcard search
+		opts.Hosts.Values = append(opts.Hosts.Values, "*")
+		routes, err = cfc.Routes.ListAll(context.Background(), opts)
+		if err != nil {
+			return &resource.Route{}, err
+		}
+		if len(routes) == 0 {
+			return &resource.Route{}, fmt.Errorf("error retrieving apps: route '%s' not found", routeUrl.Hostname())
+		}
+	}
+
+	return routes[0], nil
+}
+
+func lookup(cfc *client.Client, route *resource.Route, target bool, cliConnection plugin.CliConnection) error {
+	if route.Destinations == nil || len(route.Destinations) == 0 {
+		return fmt.Errorf("route not bound to any applications")
+	}
+
+	destinationCount := 0
+	for _, destination := range route.Destinations {
+		app, err := cfc.Applications.Get(context.Background(), *destination.App.GUID)
+		if err != nil {
+			return fmt.Errorf("route not bound to any applications")
+		}
+
+		space, org, err := cfc.Spaces.GetIncludeOrganization(context.Background(), app.Relationships.Space.Data.GUID)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Bound to:\nOrganization: %s (%s)\n", org.Name, org.GUID)
+		fmt.Printf("Space       : %s (%s)\n", space.Name, space.GUID)
+		fmt.Printf("App         : %s (%s)\n", app.Name, app.GUID)
+
+		if target && destinationCount == 0 {
+			fmt.Printf("Targeting an app's organization and space...\n")
+			_, err := cliConnection.CliCommand("target", "-o", org.Name, "-s", space.Name)
+			if err != nil {
+				fmt.Printf("targeting an app's organization and space failed\n")
+			}
+			fmt.Printf("Targeting an app's organization and space successful.\n")
+		}
+		destinationCount++
```

Comment:
this is also not needed if you take the loop index variable that is currently ignored.